packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,940 @@
+		Object {
+		  "fetchExperimentAssignment": Array [
+		    Array [
+		      Object {
+		        "anonId": "the-anon-id-123",
+		        "experimentName": "experiment_name_a",
+		      },
+		    ],
+		  ],
+		  "getAnonId": Array [
+		    Array [],
+		  ],
+		  "logError": Array [],
+		}
+	` );
+	} );
+	it( `[developmentMode] should successfully load an ExperimentAssignment`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		mockedConfig.isDevelopmentMode = true;
+		const client = createExPlatClient( mockedConfig );
+		spiedMonotonicNow.mockImplementationOnce(
+			() => validExperimentAssignment.retrievedTimestamp + 1000
+		);
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( validExperimentAssignment );
+		expect( allMockedConfigCalls() ).toMatchInlineSnapshot( `
+		Object {
+		  "fetchExperimentAssignment": Array [
+		    Array [
+		      Object {
+		        "anonId": undefined,
+		        "experimentName": "experiment_name_a",
+		      },
+		    ],
+		  ],
+		  "getAnonId": Array [
+		    Array [],
+		  ],
+		  "logError": Array [],
+		}
+	` );
+	} );
+	it( `Invalid experimentName: should return fallback and log`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		mockedConfig.isDevelopmentMode = false;
+		const client = createExPlatClient( mockedConfig );
+		await expect(
+			client.loadExperimentAssignment( 'the-invalid-experiment-name' )
+		).resolves.toEqual( {
+			experimentName: 'fallback_experiment_assignment',
+			retrievedTimestamp: validExperimentAssignment.retrievedTimestamp,
+			ttl: 60,
+			variationName: null,
+			isFallbackExperimentAssignment: true,
+		} );
+		expect( allMockedConfigCalls() ).toMatchInlineSnapshot( `
+		Object {
+		  "fetchExperimentAssignment": Array [],
+		  "getAnonId": Array [],
+		  "logError": Array [
+		    Array [
+		      Object {
+		        "experimentName": "the-invalid-experiment-name",
+		        "message": "Invalid experimentName: the-invalid-experiment-name",
+		      },
+		    ],
+		  ],
+		}
+	` );
+	} );
+	it( `[developmentMode] Invalid experimentName: should throw`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		mockedConfig.isDevelopmentMode = true;
+		const client = createExPlatClient( mockedConfig );
+		await expect(
+			client.loadExperimentAssignment( 'the-invalid-experiment-name' )
+		).rejects.toThrow( 'Invalid experimentName: the-invalid-experiment-name' );
+		expect( allMockedConfigCalls() ).toMatchInlineSnapshot( `
+		Object {
+		  "fetchExperimentAssignment": Array [],
+		  "getAnonId": Array [],
+		  "logError": Array [
+		    Array [
+		      Object {
+		        "experimentName": "the-invalid-experiment-name",
+		        "message": "Invalid experimentName: the-invalid-experiment-name",
+		      },
+		    ],
+		  ],
+		}
+	` );
```

Comment:
This is a good example of a case where we really shouldn't care about the mocks called. Eliminating such cases would make this big test file much easier to digest and maintain.

I'll hold off on reading the rest until we've discussed the points made so far.

Reply:
If the size of the file is a concern I can split it by the top level describes.

Reply:
It's just a lot to review &ndash; I don't think we need to split it to files, just ensure that the tests only test what they need to test. Having less code to read/maintain is a nice side effect. :slightly_smiling_face: 

My point still stands: In this case, we don't care about the mocks since an error was thrown, so I think we should remove the mock check. This applies to other cases where the behaviour is enough to tell us that the mocks were used as expected.

We can get a second opinion on this from people who work in testing, though.